pyaedt/edb_grpc/core/edb_data/components_data.py

```diff
@@ -0,0 +1,962 @@
+                    break
+        if len(pinNames) == self.numpins:
+            model = self._edb.Cell.Hierarchy.SPICEModel()
+            model.SetModelPath(file_path)
+            model.SetModelName(name)
+            terminal = 1
+            for pn in pinNames:
+                model.AddTerminalPinPair(pn, str(terminal))
+                terminal += 1
+        else:  # pragma: no cover
+            logging.error("Wrong number of Pins")
+            return False
+        return self._set_model(model)
+
+    @pyaedt_function_handler
+    def assign_s_param_model(self, file_path, name=None, reference_net=None):
+        """Assign S-parameter to this component.
+
+        Parameters
+        ----------
+        file_path : str
+            File path of the S-parameter model.
+        name : str, optional
+            Name of the S-parameter model.
+
+        Returns
+        -------
+
+        """
+        if not name:
+            name = get_filename_without_extension(file_path)
+
+        edbComponentDef = self.edbcomponent.GetComponentDef()
+        nPortModel = self._edb.Definition.NPortComponentModel.FindByName(edbComponentDef, name)
+        if nPortModel.IsNull():
+            nPortModel = self._edb.Definition.NPortComponentModel.Create(name)
+            nPortModel.SetReferenceFile(file_path)
+            edbComponentDef.AddComponentModel(nPortModel)
+
+        model = self._edb.Cell.Hierarchy.SParameterModel()
+        model.SetComponentModelName(name)
+        if reference_net:
+            model.SetReferenceNet(reference_net)
+        return self._set_model(model)
+
+    @pyaedt_function_handler
+    def assign_rlc_model(self, res, ind, cap, is_parallel=False):
+        """Assign RLC to this component.
+
+        Parameters
+        ----------
+        res : int, float
+            Resistance.
+        ind : int, float
+            Inductance.
+        cap : int, float
+            Capacitance.
+        is_parallel : bool, optional
+            Whether it is a parallel or series RLC component. The default is ``False``.
+        """
+        res, ind, cap = self._get_edb_value(res), self._get_edb_value(ind), self._get_edb_value(cap)
+        model = self._edb.Cell.Hierarchy.PinPairModel()
+
+        pin_names = list(self.pins.keys())
+        for idx, i in enumerate(np.arange(len(pin_names) // 2)):
+            pin_pair = self._edb.Utility.PinPair(pin_names[idx], pin_names[idx + 1])
+            rlc = self._edb.Utility.Rlc(res, True, ind, True, cap, True, is_parallel)
+            model.SetPinPairRlc(pin_pair, rlc)
+        return self._set_model(model)
+
+    @pyaedt_function_handler
+    def create_clearance_on_component(self, extra_soldermask_clearance=1e-4):
+        """Create a Clearance on Soldermask layer by drawing a rectangle.
+
+        Parameters
+        ----------
+        extra_soldermask_clearance : float, optional
+            Extra Soldermask value in meter to be applied on component bounding box.
```

Comment:
```suggestion
            Extra soldermask value in meters to apply on the component's bounding box.
```
Is meters the correct unit?